### DISCUSSION TITLE
Avalara Plugin calculates wrong taxes for order with VOUCHER applied

### DISCUSSION THREAD

**Discussion by DidoMarchet** [saleor/saleor#10117] [General] [answered]
Hi @IKarbowiak  and @timuric 
we faced the same issue I noticed in the past (solved https://github.com/saleor/saleor/discussions/9682) but this time with the VOUCHER in Saleor dashboard v3.1.0-dev core v3.3.1

Basically If a customer applies a voucher to an order, in the Saleor checkout taxes are calculated as aspected but when the order  is sent to Avalare, Saleor gives the undiscounted price instead of the reduced price.

This is the order discounted in the Saleor CP

<img width="851" alt="Screenshot 2022-07-07 at 11 34 41" src="https://user-images.githubusercontent.com/38940340/177742725-c4969b6f-3e93-456a-9579-234a1f6236be.png">

And this is in Avalara. Salero pass 539.99 instead of 539.99 - 161.99 (the voucher value)

<img width="1522" alt="Screenshot 2022-07-07 at 11 34 26" src="https://user-images.githubusercontent.com/38940340/177743057-367988f4-f06b-4fb9-9acd-261822d48b01.png">

Let me know if you need more info or If you can provide a fix (I think is similar to the old issue) because taxes not calculated for big voucher could be annoying.

Thanks in advance and kind regards,

Davide

**Comment by karolkielecki** [comment, marked as answer]
Hi @DidoMarchet, have you checked with 3.4? The following fix, mentioned in the previous thread, was introduced to Saleor 3.4 https://github.com/saleor/saleor/pull/9633

**Reply by DidoMarchet** [reply]
Hi @karolkielecki actually we are working on this previous version but I'll notify soon to update.
I let you know and if the issue is fixed I let you know.

Thanks to all and regards,

Davide

**Reply by DidoMarchet** [reply]
> Hi @DidoMarchet, have you checked with 3.4? The following fix, mentioned in the previous thread, was introduced to Saleor 3.4 #9633

Hi It works as aspected! Sorry for the topic!

Thanks and regards,

Davide

**Reply by karolkielecki** [reply]
Great! Cheers!
